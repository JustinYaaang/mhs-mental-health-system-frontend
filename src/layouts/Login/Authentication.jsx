import React from 'react'
import 'assets/css/LoginForm.css'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as sessionActions from '../../actions/SessionActions'
import Button from 'components/CustomButtons/Button.jsx'

class Authentication extends React.Component {
  constructor (props) {
    super(props)
    this.state = { credentials: { email: '', password: '' } }
    this.onChange = this.onChange.bind(this)
    this.onSave = this.onSave.bind(this)
  }

  onChange (event) {
    const field = event.target.name
    const credentials = this.state.credentials
    credentials[field] = event.target.value
    return this.setState({ credentials: credentials })
  }

  onSave (event) {
    var flag = true
    if (this.state.credentials.email === '') {
      document.getElementById('maillabel').innerHTML = 'Please fill your email address'
      document.getElementById('mailInput').style.backgroundColor = '#FEC2C2'
      flag = false
    }
    if (this.state.credentials.password === '') {
      document.getElementById('passlabel').innerHTML = 'Please fill your password'
      document.getElementById('passwordInput').style.backgroundColor = '#FEC2C2'
      flag = false
    }
    if (flag) {
      document.getElementById('passwordInput').style.backgroundColor = 'white'
      document.getElementById('mailInput').style.backgroundColor = 'white'
      event.preventDefault()
      var auth = this.props.actions.logInUser(this.state.credentials)
      auth.then(response => {
        if (!response) {
          document.getElementById('mainlabel').style.color = 'red'
          document.getElementById('mainlabel').innerHTML = 'Wrong Username/Password'
        }
      })
    }
  }

  render () {
    return (
      <form class='loginform'>
        <div class='form-group'>
          <label className='label-title' for='label'>Welcome to MHS</label>
          <label className='label-subtitle' id='mainlabel' for='label'>Please enter your login credentials.</label>
          <br />
          <label className='label-header' id='maillabel' for='exampleInputEmail1'>Email address</label>
          <input name='email' type='email' class='form-control' id='mailInput' aria-describedby='emailHelp' placeholder='Enter email' onChange={this.onChange} />
        </div>
        <div class='form-group'>
          <label className='label-header' id='passlabel' for='exampleInputPassword1'>Password</label>
          <input name='password' type='password' class='form-control' id='passwordInput' placeholder='Enter Password' onChange={this.onChange} />
        </div>
        <Button onClick={this.onSave} id='loginbutton' type='button' color='primary'>Login</Button>
      </form>
    )
  }
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators(sessionActions, dispatch)
  }
}

export default connect(null, mapDispatchToProps)(Authentication)
