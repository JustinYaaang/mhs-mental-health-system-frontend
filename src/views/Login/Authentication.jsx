import React from 'react'
// nodejs library to set properties for components
import PropTypes from 'prop-types'
// react plugin for creating charts
import ChartistGraph from 'react-chartist'
// @material-ui/core
import withStyles from '@material-ui/core/styles/withStyles'
import Icon from '@material-ui/core/Icon'
// @material-ui/icons
import DateRange from '@material-ui/icons/DateRange'
import LocalOffer from '@material-ui/icons/LocalOffer'
import Update from '@material-ui/icons/Update'
import ArrowUpward from '@material-ui/icons/ArrowUpward'
import AccessTime from '@material-ui/icons/AccessTime'
import Accessibility from '@material-ui/icons/Accessibility'
import Grade from '@material-ui/icons/Grade'
import Code from '@material-ui/icons/Code'
import All from '@material-ui/icons/AllInboxOutlined'
// core components
import GridItem from 'components/Grid/GridItem.jsx'
import GridContainer from 'components/Grid/GridContainer.jsx'
import Tasks from 'components/Tasks/Tasks.jsx'
import CustomTabs from 'components/CustomTabs/CustomTabs.jsx'
import Card from 'components/Card/Card.jsx'
import Button from 'components/CustomButtons/Button.jsx'
import CardHeader from 'components/Card/CardHeader.jsx'
import CardIcon from 'components/Card/CardIcon.jsx'
import CardBody from 'components/Card/CardBody.jsx'
import CardFooter from 'components/Card/CardFooter.jsx'
import swal from 'sweetalert'
import InformationCard from 'components/DashboardComponent/InformationCard.jsx'
import LineGraph from 'components/DashboardComponent/LineGraph.jsx'
import CustomInput from 'components/CustomInput/CustomInput'
import 'assets/css/LoginForm.css'
class Authentication extends React.Component {
  // this.loginNow=this.loginNo w.bind(this)

  loginNow () {
    if (document.getElementById('mailInput').value === '' || document.getElementById('passwordInput').value === '') {
      console.log('empty')
    }
  }

  render () {
    const { classes } = this.props

    return (
      <form class='loginform'>
        <div class='form-group'>
          <label class='label-title' for='label'>Welcome to MHS</label>
          <label class='label-subtitle' for='label'>Please enter your login credentials.</label>
          <br />
          <label class='label-header' for='exampleInputEmail1'>Email address</label>
          <input type='email' class='form-control' id='mailInput' aria-describedby='emailHelp' placeholder='Enter email' />
          {/* <small id='emailHelp' class='form-text text-muted'>We'll never share your email with anyone else.</small> */}
        </div>
        <div class='form-group'>
          <label class='label-header' for='exampleInputPassword1'>Password</label>
          <input type='password' class='form-control' id='passwordInput' placeholder='Enter Password' />
        </div>
        <Button
          onClick={() => { this.loginNow() }}
          id='loginbutton' type='button' color='primary'>Login</Button>
      </form>
    )
  }
}

export default Authentication
