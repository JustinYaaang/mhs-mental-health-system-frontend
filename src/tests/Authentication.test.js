import React from 'react'
import { shallow } from 'enzyme'
import Authentication from '../views/Login/Authentication'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe('SurveyCreator', () => {
    it('should render correctly', async() => {
        const component = shallow( < Authentication / > )

        expect(component).toMatchSnapshot()
    })
})