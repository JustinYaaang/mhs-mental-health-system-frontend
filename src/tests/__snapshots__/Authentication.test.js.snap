// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyCreator should render correctly 1`] = `
<form
  class="loginform"
>
  <div
    class="form-group"
  >
    <label
      class="label-header"
      for="exampleInputEmail1"
    >
      Email address
    </label>
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="exampleInputEmail1"
      placeholder="Enter email"
      type="email"
    />
  </div>
  <div
    class="form-group"
  >
    <label
      class="label-header"
      for="exampleInputPassword1"
    >
      Password
    </label>
    <input
      class="form-control"
      id="exampleInputPassword1"
      placeholder="Password"
      type="password"
    />
  </div>
  <div
    class="form-check"
  >
    <label
      class="form-check-label"
    >
      <input
        class="form-check-input"
        type="checkbox"
        value=""
      />
      Option one is this
      <span
        class="form-check-sign"
      >
        <span
          class="check"
        />
      </span>
    </label>
  </div>
  <button
    class="btn btn-primary"
    type="submit"
  >
    Submit
  </button>
</form>
`;
